src/subscript/welltest_extract/welltest_extract.py

```diff
@@ -0,0 +1,396 @@
+    )
+    parser.add_argument(
+        "out_files_prefix",
+        type=str,
+        help="Prefix to be added to result files, can contain a path",
+    )
+    parser.add_argument(
+        "-n",
+        "--buildup_nr",
+        type=int,
+        help="Buildup number, indicating which build up to extract. Counting from 1 ",
+        default=1,
+        required=False,
+    )
+    parser.add_argument(
+        "--phase",
+        type=str,
+        help="Main fluid phase in test (OIL/GAS)",
+        default="OIL",
+        required=False,
+    )
+    return parser
+
+
+def get_summary_vec(summary, key):
+    """
+    Read vector corresponding to key from summary instance
+
+    Return:
+    vec  : np.array
+    """
+
+    try:
+        return summary.numpy_vector(key)
+    except KeyError:
+        print("No such key in summary file:" + key)
+        raise
+
+
+def get_buildup_indices(rates):
+    """
+    Go through the simulated rate and identify bu periods, as defined by zero flow.
+
+    Returns:
+    buildup_incices     : list of indices associated with start of the build ups
+    buildup_end_incices : list of indices associated with end of the build ups
+
+    """
+
+    buildup_indices = []
+    buildup_end_indices = []
+
+    last = 0
+    for i, rate in enumerate(rates):
+        if rate == 0 and last > 0.0:
+            buildup_indices.append(i)
+        if rate > 0 and last == 0:
+            buildup_end_indices.append(i - 1)
+        if i == len(rates) - 1 and rate == 0:
+            buildup_end_indices.append(i)
+        last = rate
+
+    if rates[0] == 0:
+        del buildup_end_indices[0]
+
+    return buildup_indices, buildup_end_indices
+
+
+def get_supertime(time, rate, bu_start_ind, bu_end_ind):
+    """
+    Calculate supertime as
+
+    Return: supertime - np.array
+    """
+
+    rdiff = np.diff(rate)
+    rdiff = np.hstack((rate[0] - 0, rdiff))
+
+    super_time = np.zeros(bu_end_ind + 1 - bu_start_ind)
+    coeff1 = 1 / (0.0 - rate[bu_start_ind - 1])
+    # 1/(q_n - q_n-1)  where q_n is zero (start of BU and q_n-1 is last rate before BU
+
+    for bu_time_ind in range(1, bu_end_ind - bu_start_ind + 1):
+        # Cannot start from zero. Hence from 1 and not 0 in loop. (Avoid ln(0))
+        tot = 0.0
+        for i in range(0, bu_start_ind):
+            # End at len-1 because n is not included - only n-1 in formula
+            tot = tot + rdiff[i] * np.log(time[bu_start_ind + bu_time_ind] - time[i])
+        super_time[bu_time_ind] = coeff1 * tot + np.log(
+            time[bu_start_ind + bu_time_ind] - time[bu_start_ind]
+        )
+
+    return super_time[1:]
+
+
+def get_weighted_avg_press_time_derivative_lag1(dp, dspt):
+    """
+    Compute weighted average of pressure time derivative,
+    one time step to each side. Lag1
+
+    Formula: (  (dp_f/dspt_f)*dspt_b + (dp_b/dspt_b)*dspt_f )/(dspt_f + dspt_b)
+
+    spt is SuperPositionedTime and dspt is delta spt
+    """
```

Comment:
For later refactoring, we need to know the types of the function arguments. Add to docstring?

Reply:
Agree, docstrings updated all over